src/ast/codegen_llvm.cpp

```diff
@@ -1625,7 +1625,30 @@ void CodegenLLVM::visit(Tuple &tuple)
 {
   // Store elements on stack
   llvm::Type *tuple_ty = b_.GetType(tuple.type);
+
+  // Validate that what we thought the struct looks like
+  // and LLVM made of it are equal to avoid issues.
+  size_t struct_size = layout_.getTypeAllocSize(tuple_ty);
+  if (struct_size != tuple.type.size)
+  {
+    LOG(FATAL) << "Struct size mismatch: " << tuple.type.size
+               << " != " << struct_size;
+  }
+
+  auto *struct_layout = layout_.getStructLayout(
+      reinterpret_cast<llvm::StructType *>(tuple_ty));
+  for (ssize_t i = 0; i < tuple.type.GetFieldCount(); i++)
+  {
+    ssize_t offset = struct_layout->getElementOffset(i);
+    if (tuple.type.GetField(i).offset != offset)
+    {
+      LOG(FATAL) << "Struct offset mismatch: " << i << ": " << tuple.type.size
```

Comment:
Offset mismatch (wrong vars here too), shouldn't be fatal as the offset is only relevant when printing. So we can just fix the struct. As long as the size matches it's ok